test/Test/Extensions.hs

```diff
@@ -2,38 +2,82 @@ module Test.Extensions
     ( getExtensionsSpec
     ) where
 
+import Data.ByteString (ByteString)
 import Data.Map.Strict (Map)
+import Data.Maybe (fromMaybe)
 import Data.Set (Set)
 import GHC.LanguageExtensions.Type (Extension (..))
 import Test.Hspec (Spec, describe, it, shouldBe)
 
-import Extensions (ExtensionsResult, getPackageExtentions)
+import Extensions (ExtensionsError (..), ExtensionsResult, getModuleExtentions,
+                   getModuleExtentionsBySource, getPackageExtentions, getPackageExtentionsBySources)
 import Extensions.OnOff (OnOffExtension (..))
 import Test.Extensions.Cabal (defaultExtensions)
 
+import qualified Data.ByteString as BS
 import qualified Data.Map.Strict as Map
 import qualified Data.Set as Set
 
 
 getExtensionsSpec :: Spec
-getExtensionsSpec = describe "Get Extensions" $
-    it "should get package extensions" $
-        getPackageExtentions "extensions.cabal" >>= \res -> res `shouldBe` resultMap
+getExtensionsSpec = describe "Get Extensions" $ do
+    it "should getPackageExtensions" $
+        getPackageExtentions cabal >>= \res -> res `shouldBe` resultMap
+    it "should getPackageExtensionsBySource" $ do
+        res <- sources >>= getPackageExtentionsBySources cabal
+        res `shouldBe` resultMap
+    it "should getPackageExtensionsBySource, return NonCabalModule" $ do
+        srcs <- sources
+        res <- getPackageExtentionsBySources cabal $
+            Map.insert unknownModule "Hello" srcs
+        res `shouldBe` Map.insert unknownModule (Left $ NotCabalModule unknownModule) resultMap
```

Comment:
What do you think about inversing how the comparison done in this particular case? I guess instead of comparing two `Map`s, we can compare two values. The current implementation is correct, but I'm afraid that the diff in the test output will be big in case of test failure. So instead of:

```haskell
res `shouldBe` someMap
```

we can

```haskell
Map.lookup unknownModule res `shouldBe` Just ...
```

What do you think?

Reply:
Interesting approach! I see how it could help to improve some tests.
At the moment I would keen to keep the current comparison with Maps, as it doesn't take too much time for now and also it could inform us if we change something that could break things for other elements in the Map too. Do you agree to keep it as it is for now and reconsider when it brings troubles?

Reply:
@vrom911 Good point! I haven't thought about that.